### ISSUE TITLE
geodataframe.to_crs results in inf

### ISSUE THREAD

**Issue by wassname** [geopandas/geopandas#798]
I found a polygon that results in infinite values when converted to epsg 3854. Here's a minimal example:

```py
vectorff = '/tmp/vec.json'
open(vectorff, 'w').write('{"type": "FeatureCollection", "features": [{"id": "BRN.1_1", "type": "Feature", "properties": {"GID_0": "BRN", "NAME_0": "Brunei", "NAME_1": "Belait"}, "geometry": {"type": "Polygon", "coordinates": [[[114.82219696, 4.17134094], [114.62239838, 4.02623987], [114.12805176, 4.5869441], [114.52954865, 4.71593618], [114.87163544, 4.34088421], [114.82219696, 4.17134094]]]}}, {"id": "LAO.8_1", "type": "Feature", "properties": {"GID_0": "LAO", "NAME_0": "Laos", "NAME_1": "Louangphrabang"}, "geometry": {"type": "Polygon", "coordinates": [[[102.15626526, 19.40581894], [101.80967712, 19.06066895], [101.89360809, 19.41581345], [101.71374512, 19.91818619], [102.19501495, 20.14537048], [102.30948639, 20.96255875], [102.49090576, 20.85570717], [102.94989014, 21.14259911], [103.38887024, 20.78600311], [103.21352386, 20.69704819], [103.13821411, 19.91510773], [102.66527557, 19.51914787], [102.70091248, 19.27285957], [102.15626526, 19.40581894]]]}}]}')
gpd.read_file(vectorff).to_crs(epsg=3854)
```

Ouput:

|| GID_0 | NAME_0 | NAME_1 | geometry|
|-- | -- | -- | -- | -- | 
|BRN.1_1 | BRN | Brunei | Belait | POLYGON ((inf inf, inf inf, inf inf, inf inf, ...|
|LAO.8_1 | LAO | Laos | Louangphrabang | POLYGON ((12238091.72200898 1466459.78213906, ...|

- versions: fiona-1.7.13 geopandas-0.4.0 munch-2.3.2 numpy-1.15.0 pandas-0.23.4 python-dateutil-2.7.3 pytz-2018.5 shapely-1.6.4.post2
- python: 3.6

Is this an issue for anyone else, and does anyone know what causes it?

**Comment by cipri-tom** [comment]
@jorisvandenbossche I would like to reopen this issue, as I am facing it with two polygons that should be both valid for the given projection, yet one fails to be projected, resulting in `inf`.

```python
gpd.read_file('inf.geojson')\
    .to_crs(crs='+proj=lcc +lat_1=33 +lat_2=45 +lat_0=39 +lon_0=-96 +x_0=0 +y_0=0 +ellps=GRS80 +datum=NAD83 +units=m +no_defs')
```

The CRS is taken from [here](https://spatialreference.org/ref/esri/usa-contiguous-lambert-conformal-conic/), recommended for projecting continental US.

Any insight why ?
Thanks !

**Comment by martinfleis** [comment]
@cipri-tom I can't reproduce the issue. Can you ensure that you're using the latest versions of geopandas and pyproj? If it still fails, can you open a new issue?

**Comment by RJbalikian** [comment]
I am having a similar issue when trying to project between EPSG:4269 and EPSG:26716 and/or EPSG:26715 (I have data all over the state of Illinois/which spans two UTM zones, so I may need to use both of these). I get all Point (inf inf) for all records.

These are both UTM based, and the data I have spans these zones 15 (26715) and zone 16 (26716). However, when I used another UTM based CRS (EPSG:6344) it works for the points that are within the UTM zone, but outside the zone it gets the same results (Point (inf inf)). I would be fine with only having data within the UTM zone work, but not having any of it work is no good. I also tried my geodataframe with WGS84 (EPSG:4326) just to double check my code again, and it seems to work fine there no problem too. Is there something about these particular CRS?

Link below contains the outputs of what I have done. The Latitude and Longitude columns contain the original data (in 4269). I converted those to the geometry format (not shown, but no issue), then tried to reproject them to the CRS with the EPSG noted in the filenames, with the result being the geometry column. Both the 26715 and 6344 datasets are UTM Zone 15 data, so I thought that might be a good comparison. The original data is quite large (several hundred thousand points) so I have only included the first 5000 points. This is enough to show you data in both zones.

Also including a rough plot of all the data (concentrating on Illinois), with green being the data that worked (for the 6344 transform) and purple being the data that did not (i.e., that returned Point (inf inf)).

![image](https://user-images.githubusercontent.com/46536937/211646514-36132198-6889-465f-b59f-2db8245dc963.png)

[pointDataExamples.zip](https://github.com/geopandas/geopandas/files/10386215/pointDataExamples.zip)
